lua/mason-core/installer/managers/std.lua

```diff
@@ -108,7 +108,11 @@ end
 local function untar(rel_path)
     log.fmt_debug("std: untar %s", rel_path)
     local ctx = installer.context()
-    return ctx.spawn.tar({ "--no-same-owner", "-xvf", rel_path }):on_success(function()
+    local tar_cmd = ctx.spawn.tar
+    if platform.is.unix and not platform.is.linux then
+        tar_cmd = ctx.spawn.gtar
+    end
```

Comment:
Sections like this with the platform check would be nice to refactor into a function somewhere else (but I don't know how you would want to organize this so I've left them as is for now). I see this as being useful because if the code needed to change for some reason, it could be changed in one place.